aten/src/ATen/core/functional.h

```diff
@@ -13,14 +13,23 @@ namespace c10 {
 //
 // No explicit template parameters are required.
 
+
+// A traditional C++ approach with iterators
+template<class F, class InputIt>
+inline auto fmap(InputIt first, InputIt last, const F& fn)
+            -> std::vector<decltype(fn(*first))> {
+  std::vector<decltype(fn(*first))> r;
```

Comment:
reserve space in `r` like line 20 of original code to save memory allocation time.

Reply:
Done, but please see comment above about `std::distance` issues